extension/src/extension.ts

```diff
@@ -432,7 +446,35 @@ export class Extension extends Disposable implements IExtension {
 let extension: undefined | Extension
 export function activate(context: ExtensionContext): void {
   extension = new Extension(context)
-  context.subscriptions.push(extension)
+  const dvcYamlSelector = {
+    language: 'yaml'
+  }
+
+  context.subscriptions.push(
+    extension,
+    languages.registerDocumentLinkProvider(
+      dvcYamlSelector,
+      new DvcYamlDocumentLinkProvider()
+    ),
+    languages.registerCompletionItemProvider(
+      dvcYamlSelector,
+      new DvcYamlCompletionProvider(),
+      '.',
+      '/'
+    ),
+    languages.registerHoverProvider(
```

Comment:
These extra providers will be removed for this PR

Reply:
[Q] Did you look at `registerDefinitionProvider`? Might be able to use that to step into a file?

Reply:
or you were planning to use `registerDocumentSymbolProvider`?

Reply:
Both, actually, I think these definition/symbol providers will be easier to do